File: backend/booking-service/src/main/java/com/tripify/booking_service/dto/BookingResponseDTO.java
package com.tripify.booking_service.dto;

import com.tripify.booking_service.entity.BookingStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

public record BookingResponseDTO(
        Long id,
        BigDecimal totalAmount,
        LocalDateTime bookingDate,
        BookingStatus status,

        // IL CAMPO MAGICO PER IL FRONTEND
        boolean isLeader,

        List<String> participantIds,
        List<BookingLineDTO> lines
) {}

File: frontend/app/src/main/java/com/tripify/tripify_android/data/model/BookingResponseDTO.kt
package com.tripify.tripify_android.data.model

data class BookingResponseDTO(
    val id: Long,
    val totalAmount: Double,
    val bookingDate: String,
    val status: String,
    val isLeader: Boolean,
    val participantIds: List<String> = emptyList(),
    val lines: List<BookingLineDTO> = emptyList()
)
